### PR TITLE
Move everything from testify/assert to require

### PR DESCRIPTION
**Pull request by sykesm**
It’s time we leave `assert` behind and make a complete transition to `require`. I haven’t yet run into a situation where we explicitly want a test to proceed after an assertion failure.

- apply `gofmt` rewriting rules for assert -> require
- replace non-asserting `assert.ObjectsAreEqual` with alternative assertions
- update doc to call out `testify/require` instead of `testify/assert`

### CODE REVIEW

**Comment by btl5037** [comment]
image

**Comment by lindluni** [comment]
Wow. This was as aweful reviewing as it looks like it would be. But thank you for doing this. Will actually make debugging failures easier as they won't be wrapped in output of other tests.

**Review by lindluni — APPROVED** [review]

File: core/common/ccprovider/ccinfocache_test.go [hunk 1]
```diff
@@ -241,8 +241,8 @@ func TestSetChaincodesPath(t *testing.T) {
 	defer SetChaincodesPath(cip)
 
 	f, err := ioutil.TempFile(dir, "chaincodes")
-	assert.NoError(t, err)
-	assert.Panics(t, func() {
+	require.NoError(t, err)
+	require.Panics(t, func() {
```
Comment (cendhu) [inline core/common/ccprovider/ccinfocache_test.go:245]: Unrelated to this PR. Line 249 to 263 needs to be fixed in a separate PR.
